### PR TITLE
fix(pagination): fixes pagination on datasets, registry datasets, profiles, search results, and history

### PR DESCRIPTION
**Pull request by ramfox**
- adds selector for `PageCount` in the Registry list paginator.
- adds eventlistener to app component, when the app scrolls, sees if the user has scrolled to the bottom of the app component. if so, set's `atBottom` true. If `atBottom` was previously true, and we are no longer at the bottom of the app component, set's `atBottom` to false
- uses this new `atBottom` val in the List component to determine if we should load the next page
- fixes the `fetchPublishedDatasets` function, although tbh it's not being used anywhere at the moment
- sets up `Datasets`, `Network`, `SearchResults`, and `Commits` to have infinite scroll!!!

### CODE REVIEW

File: lib/components/App.js [hunk 1]
```diff
@@ -115,6 +118,13 @@ export default class App extends React.PureComponent {
     this.props.hideModal()
   }
 
+  handleMainScroll () {
+    const { atBottom } = this.props
+    if (atBottom !== (this.main.current.scrollHeight === this.main.current.scrollTop + this.main.current.offsetHeight)) {
```
Comment (b5) [inline lib/components/App.js]: There are often slightly scary implications that always show up when hooking into scroll events (they're called _lots_). In this case we have code:
* a function called on all scroll events 
* that can affect the state tree, 
* which can cause a repaint (calling render again) 
* which can affect the height of the page
* which can invalidate the value `atBottom`

based on this, I think the more accurate name for this value is `hitBottom`, meaning _at some point_ the user scrolled all the way down.

Other than a variable rename, I think this code is pretty close to what _must_ be done to get fetching-in-reaction-to-scrolling-style behaviour. Nice work

**Review by b5 — APPROVED** [review]
purely on the basis that this adds vital functionality, I think we should merge this & deal with any cleanup later. If you can deal with the `atBottom` vs `hitBottom` thing, great!

File: lib/components/App.js (hunk 1 above)
Comment (ramfox) [reply lib/components/App.js]: I think that if i've learned anything in the past two reviews its that I'm terrible at names XD

Comment (b5) [reply lib/components/App.js]: no no don't think that! Naming is _really, really, really_ hard. It's also opinionated 😄

**Pull request merged**